classes/submissionFile/Repository.php

```diff
@@ -843,4 +859,262 @@ protected function getSubmissionFileLogData(SubmissionFile $submissionFile): arr
             'username' => $user?->getUsername(),
         ];
     }
+
+    /**
+     * Add jatsContent for Submission files that correspont to 
+     */
+    public function getJatsFileProps(SubmissionFile $submissionFile, array $genres): array
+    {
+        $fileProps = $this->getSchemaMap()->summarize($submissionFile, $genres);
+        $jatsFileName = Config::getVar('files', 'files_dir') . '/' . $fileProps['path'] .'';
+        $xmlString = file_get_contents($jatsFileName);
+
+        if ($xmlString) {
+            $fileProps['jatsContent'] = $xmlString;
+        }
+
+        return $fileProps;
+    }
+
+    /**
+     * Returns the SubmissionFile, if any, that corresponds to the JATS contents of the given submission/publication
+     */
+    public function getJatsFiles(int $submissionId, int $publicationId): ?SubmissionFile
+    {
+        $this->validateSubmissionAndPublication($submissionId, $publicationId);
+
+        $submissionFile = Repo::submissionFile()
+            ->getCollector()
+            ->filterBySubmissionIds([$this->submission->getId()])
+            ->filterByFileStages([SubmissionFile::SUBMISSION_FILE_JATS])
+            ->filterByAssoc(Application::ASSOC_TYPE_PUBLICATION, [$this->publication->getId()])
+            ->getMany()
+            ->first();
+        
+        return $submissionFile;
+    }
+
+    /**
+     * Returns the name of the file that will contain the default JATS content 
+     */
+    public function getDefaultJatsFileName(int $submissionId, int $publicationId): string
+    {
+        return 'jats-' . $submissionId . '-' . $publicationId . '-' . (string) mt_rand(100000, 999999) . '.xml';
+    }
+
+    /**
+     * Creates the default JATS XML Content from the given submission/publication metadata
+     */
+    public function createDefaultContentJatsFile(int $submissionId, int $publicationId): string
+    {
+        $submission = Repo::submission()->get($submissionId);
```

Comment:
I think this will already be removed, but wherever possible it's a good idea to ensure that your IDs match appropriately where it's possible to specify several. That would mean either simplifying this down to only accept a `$publicationId`, or making sure that `$publicationId` and `$submissionId` are *both* passed in to the `Repo::publication()::get` function.

Reply:
passed both `publicationId` and `$submissionId` in the getter.